js/views/reader_view.js

```diff
@@ -519,58 +519,49 @@ ReadiumSDK.Views.ReaderView = function(options) {
      * @fires ReadiumSDK.Events.SETTINGS_APPLIED
      */
     this.updateSettings = function(settingsData) {
-
-//console.debug("UpdateSettings: " + JSON.stringify(settingsData));
-
-        _viewerSettings.update(settingsData);
-        
-        if (_mediaOverlayPlayer)
-        {
-            _mediaOverlayPlayer.setAutomaticNextSmil(_viewerSettings.mediaOverlaysAutomaticPageTurn ? true : false);
-        }
-        
-        if(_currentView && !settingsData.doNotUpdateView) {
-
-            var bookMark = _currentView.bookmarkCurrentPage();
-
-            if(bookMark && bookMark.idref) {
-     
-                var wasPlaying = false;
-                if (_currentView.isReflowable && _currentView.isReflowable())
-                {
-                    wasPlaying = self.isPlayingMediaOverlay();
-                    if (wasPlaying)
-                    {
-                        self.pauseMediaOverlay();
-                    }
-                }
-
-                var spineItem = _spine.getItemById(bookMark.idref);
-                
-                initViewForItem(spineItem, function(isViewChanged){
-
-                    if(!isViewChanged) {
-                        _currentView.setViewSettings(_viewerSettings);
-                    }
-
-                    self.openSpineItemElementCfi(bookMark.idref, bookMark.contentCFI, self);
-
-                    if (wasPlaying)
-                    {
-                        self.playMediaOverlay();
-                        // setTimeout(function()
-                        // {
-                        // }, 60);
-                    }
-
-                    self.trigger(ReadiumSDK.Events.SETTINGS_APPLIED);
-                    return;
```

Comment:
Moving the return is the simpler fix. My normal inclination is not to ```return``` from deeply nested code, but the ```return``` is cleaner as opposed to the multiple triggers in this case. I will update the PR.